src/System.Device.Gpio/System/Device/Gpio/Drivers/LibGpiodDriver.Linux.cs

```diff
@@ -137,22 +273,41 @@ protected internal override void Write(int pinNumber, PinValue value)
 
         protected override void Dispose(bool disposing)
         {
-            foreach (int pin in _pinNumberToSafeLineHandle.Keys)
+            _disposing = disposing;
+
+            if (_pinNumberToEventHandler != null)
             {
-                if (_pinNumberToSafeLineHandle.TryGetValue(pin, out SafeLineHandle pinHandle))
+                foreach (var kv in _pinNumberToEventHandler)
                 {
-                    Interop.ReleaseGpiodLine(pinHandle);
+                    int pin = kv.Key;
+                    LibGpiodDriverEventHandler eventHandler = kv.Value;
+                    eventHandler.Dispose();
```

Comment:
I'm wondering if event handler should own the pin here - that suggest that if I register for the event and use regular APIs at the same time and then unregister the event then I will find myself with the disposed pin?

Reply:
I am wondering if that scenario could work for the controller. I have got pin not available error when trying to listening event for a pin set for Input or Output. With that i thinkg one cannot use a pin for regular pin API and event handling simultaniosly. I didn't exactly tested your scenario though ...

Reply:
that's actually not something I'd expect but if that is the limitation perhaps it might be worth throwing when you try to do both simulatanously